resources/user.go

```diff
@@ -0,0 +1,275 @@
+	log.Printf("%s: CheckApply(%t)", obj, apply)
+
+	var exists = true
+
+	usr, err := user.Lookup(obj.GetName())
+	if err != nil {
+		if err == err.(user.UnknownUserError) {
+			log.Printf("the user: %s does not exist", obj.GetName())
+			exists = false
+		} else {
+			return false, fmt.Errorf("error looking up user")
+		}
+	}
+
+	if obj.AllowDuplicateUID == false {
+		existingUID, err := user.LookupId(strconv.Itoa(int(*obj.UID)))
+		if err == nil && existingUID.Username != obj.GetName() {
+			return false, fmt.Errorf("the uid: %d is already taken", *obj.UID)
+		}
+	}
+
+	if exists && obj.State == "exists" {
+		usercheck := true
+		if obj.GID != nil && strconv.Itoa(int(*obj.GID)) != usr.Gid {
+			usercheck = false
+		}
+		if obj.UID != nil && strconv.Itoa(int(*obj.UID)) != usr.Uid {
+			usercheck = false
+		}
+		if obj.HomeDir != "" && obj.HomeDir != usr.HomeDir {
+			usercheck = false
+		}
+		if usercheck {
+			return true, nil
+		}
+	}
+
+	if obj.State == "absent" && !exists {
+		return true, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+
+	var cmdName string
+	var args []string
+	if obj.State == "exists" {
+		if !exists {
+			cmdName = "useradd"
+			log.Printf("adding user: %s", obj.GetName())
+		}
+		if exists {
+			cmdName = "usermod"
+			log.Printf("modifying user: %s", obj.GetName())
+		}
+		if obj.AllowDuplicateUID == true {
+			args = append(args, "--non-unique")
+		}
+		if obj.UID != nil {
+			args = append(args, "-u", strconv.Itoa(int(*obj.UID)))
+		}
+		if obj.GID != nil {
+			args = append(args, "-g", strconv.Itoa(int(*obj.GID)))
+		}
+		if obj.HomeDir != "" {
+			args = append(args, "-d", obj.HomeDir)
+		}
+
+	}
+
+	if obj.State == "absent" {
+		cmdName = "userdel"
+	}
+
+	args = append(args, obj.GetName())
+
+	cmd := exec.Command(cmdName, args...)
+
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Setpgid: true,
+		Pgid:    0,
+	}
+	if err := cmd.Run(); err != nil {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// UserUID is the UID struct for UserRes.
+type UserUID struct {
+	BaseUID
+	name string
+}
+
+// UIDs includes all params to make a unique identification of this object.
+// Most resources only return one, although some resources can return multiple.
+func (obj *UserRes) UIDs() []ResUID {
+	x := &UserUID{
+		BaseUID: BaseUID{Name: obj.GetName(), Kind: obj.GetKind()},
+		name:    obj.Name,
+	}
+	return []ResUID{x}
+}
+
+// GroupCmp returns whether two resources can be grouped together or not.
+func (obj *UserRes) GroupCmp(r Res) bool {
+	_, ok := r.(*UserRes)
+	if !ok {
+		return false
+	}
+	return false // TODO: this is doable!
+	// TODO: it could be useful to group our writes and watches!
+}
+
+// Compare two resources and return if they are equivalent.
+func (obj *UserRes) Compare(r Res) bool {
+	// we can only compare UserRes to others of the same resource kind
+	res, ok := r.(*UserRes)
+	if !ok {
+		return false
+	}
+	if !obj.BaseRes.Compare(res) { // call base Compare
+		return false
+	}
```

Comment:
Add in the compare checks for the new struct fields :) (See other resources for an example... careful comparing the pointers)